### PR TITLE
[FEAT] run cluster on commit

### PR DESCRIPTION
**Pull request by raunakab**
# Overview
This PR extends the existing `run-clusters` workflow with the ability to run on any arbitrary daft-wheel that is given to it.

### CODE REVIEW

File: .github/workflows/run-cluster.yaml [hunk 1]
```diff
@@ -3,9 +3,13 @@ name: run-cluster
 on:
   workflow_dispatch:
     inputs:
+      daft_wheel_url:
+        type: string
+        description: The url pointing to a daft python-wheel for which to install and run
```
Comment (jaychia) [inline .github/workflows/run-cluster.yaml:8]: Specifically, a public https URL to the wheel directly right

**Review by jaychia — APPROVED** [review]

File: .github/workflows/run-cluster.yaml (hunk 1 above)
Comment (raunakab) [reply .github/workflows/run-cluster.yaml:8]: Yup, exactly.

File: .github/workflows/run-cluster.yaml [hunk 2]
```diff
@@ -3,9 +3,13 @@ name: run-cluster
...
+        required: false
       daft_version:
         type: string
-        description: The version of daft to install
+        description: The version of daft to install (errors if both this and `daft_wheel_url` are provided)
```
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/workflows/run-cluster.yaml:12]: Will update in a follow-up PR. Accidentally jumped the gun here.

File: .github/ci-scripts/templatize_ray_config.py [hunk 3]
```diff
@@ -59,10 +61,17 @@ class Profile:
     if args.cluster_name:
         content = content.replace(CLUSTER_NAME_PLACEHOLDER, args.cluster_name)
 
-    if args.daft_version:
-        content = content.replace(DAFT_VERSION_PLACEHOLDER, f"=={args.daft_version}")
+    if args.daft_wheel_url and args.daft_version:
+        raise Exception(
```
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/ci-scripts/templatize_ray_config.py:64]: Will update in a follow-up PR as well!

File: .github/workflows/run-cluster.yaml (hunk 1 above)
Comment (raunakab) [reply .github/workflows/run-cluster.yaml:8]: Will update the description in a follow-up PR.

Comment (raunakab) [reply .github/workflows/run-cluster.yaml:8]: Updates here: https://github.com/Eventual-Inc/Daft/pull/3473.

File: .github/workflows/run-cluster.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/workflows/run-cluster.yaml:12]: Updates here: https://github.com/Eventual-Inc/Daft/pull/3473.

File: .github/ci-scripts/templatize_ray_config.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (raunakab) [reply .github/ci-scripts/templatize_ray_config.py:64]: Updates here: https://github.com/Eventual-Inc/Daft/pull/3473.

**Pull request merged**